src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -1181,6 +1211,53 @@ private void addNewFolder() {
 		}
 	}
 
+	private void renameFolder(IElement selected) {
+		String newName = VOptionPane
+				.showInputDialog(mcreator, L10N.t("workspace.elements.folders.rename.message"), L10N.t("workspace.elements.folders.rename.title"), null, new OptionPaneValidatior() {
+					@Override public ValidationResult validate(JComponent component) {
+						String newName = ((JTextField) component).getText();
+						if (!newName.matches("[A-Za-z0-9._ -]+")) {
+							canRename = false;

```

Comment:
Why is this global variable and why do you even need this?

Reply:
For some reasons, whenever the dialog gets an error result (returning validationresult to error)
It still executes the rename with the inappropriate Input, and after that, it crashes the workspace

Reply:
Even though, it doesn't execute the action with other methods which uses same code, like AddFolder

Reply:
When validation fails, it will return null, so check for that and if it is null, either it was canceled or invalid name. Also no reason to go global with this variable even if it was needed

Reply:
I think what i need is a `if (newName != null) {`

Reply:
I think it's all good here now